src/.gitignore

```diff
@@ -0,0 +1,2 @@
+VersionInfo.h
+MQTTVersion
```

Comment:
When doing an out-of-tree build (as done in the travis build), the binaries and generated files will not clutter the source dir; as a result, I don't think there's any need to add any gitignore entry.